internal/pkg/manifest/svc.go

```diff
@@ -495,6 +495,46 @@ func (rr RoutingRuleConfiguration) exposedPorts(exposedPorts []ExposedPort, work
 	}
 }
 
+// exportPorts returns any new ports that should be exposed given the network load balancer
+// configuration that's not part of the existing containerPorts.
+func (cfg NetworkLoadBalancerConfiguration) exposedPorts(exposedPorts []ExposedPort, workloadName string) ([]ExposedPort, error) {
+	if cfg.IsEmpty() {
+		return nil, nil
+	}
+	containerName := workloadName
+	if cfg.TargetContainer != nil {
+		containerName = aws.StringValue(cfg.TargetContainer)
+	}
+	nlbPort, protocol, err := ParsePortMapping(cfg.Port)
+	if err != nil {
+		return nil, err
+	}
+	if protocol == nil {
+		protocol = aws.String("tcp")
+	}
+
+	port, err := strconv.ParseUint(aws.StringValue(nlbPort), 10, 16)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I feel like we don't need this whole thing if `cfg.TargetPort != nil`

Reply:
Do you think following is a good idea?

```
if cfg.TargetPort == nil {
  nlbPort, protocol, err := ParsePortMapping(cfg.Port)
	  if err != nil {
		  return nil, err
	  }
	  if protocol == nil {
		  protocol = aws.String("tcp")
	  }
  
	  port, err := strconv.ParseUint(aws.StringValue(nlbPort), 10, 16)
	  if err != nil {
		  return nil, err
	  }
} else {
  containerPort = uint16(aws.IntValue(cfg.TargetPort))
}
```